bibernate-core/src/test/java/com/bobocode/petros/bibernate/session/context/PersistenceContextTest.java

```diff
@@ -0,0 +1,156 @@
+
+@ExtendWith(MockitoExtension.class)
+class PersistenceContextTest {
+
+    private PersistenceContext persistenceContext;
+    private Product milk = new Product();
+    private Product banana = new Product();
+    private Product porosiatko = new Product();
+    private Order order = new Order();
+
+    @BeforeEach
+    void init() {
+        persistenceContext = new PersistenceContext();
+
+        milk.setId(1L);
+        milk.setName("Moloko");
+        milk.setPrice(BigDecimal.valueOf(2L));
+
+        porosiatko.setId(3L);
+        porosiatko.setName("Lusia");
+        porosiatko.setPrice(BigDecimal.valueOf(100));
+
+        order.setProvider("Home Farm");
+        order.setProducts(List.of(milk, banana, porosiatko));
+    }
+
+    @Test
+    void whenAddEntityToSnapshotThenItStoreInEntitySnapshotsMap() {
+        persistenceContext.addSnapshot(milk);
+        persistenceContext.addSnapshot(porosiatko);
+
+        var milkKey = new EntityKey<>(milk.getClass(), milk.getId());
+        var porosiatkoKey = new EntityKey<>(porosiatko.getClass(), porosiatko.getId());
+
+        Map<EntityKey<?>, Object> entitySnapshots = getEntityKeyToObjectMap(persistenceContext, "entitySnapshots");
+
+        assertAll(
+                () -> assertNotNull(entitySnapshots.get(milkKey)),
+                () -> assertNotNull(entitySnapshots.get(porosiatkoKey)),
+                () -> assertEquals(milk, entitySnapshots.get(milkKey)),
+                () -> assertEquals(porosiatko, entitySnapshots.get(porosiatkoKey)),
+                () -> assertTrue(entitySnapshots.containsKey(milkKey)),
+                () -> assertTrue(entitySnapshots.containsKey(porosiatkoKey)));
+    }
+
+    @Test
+    void whenAddNullOrNotEntityAsEntityToSnapshotThenThrowNullPointerException() {
+        assertAll(
+                () -> assertThrows(NullPointerException.class, () -> persistenceContext.addSnapshot(null)),
+                () -> assertThrows(NoEntityIdException.class, () -> persistenceContext.addSnapshot(order))
+        );
+    }
+
+    @Test
+    void whenAddEntityToCacheThenItStoreInCacheMap() {
+        persistenceContext.addToCache(milk);
+        persistenceContext.addToCache(porosiatko);
+
+        var milkKey = new EntityKey<>(milk.getClass(), milk.getId());
+        var porosiatkoKey = new EntityKey<>(porosiatko.getClass(), porosiatko.getId());
+
+        Map<EntityKey<?>, Object> cache = getEntityKeyToObjectMap(persistenceContext, "cache");
+
+        assertAll(
+                () -> assertNotNull(cache.get(milkKey)),
+                () -> assertNotNull(cache.get(porosiatkoKey)),
+                () -> assertEquals(milk, cache.get(milkKey)),
+                () -> assertEquals(porosiatko, cache.get(porosiatkoKey)),
+                () -> assertTrue(cache.containsKey(milkKey)),
+                () -> assertTrue(cache.containsKey(porosiatkoKey)));
+    }
+
+    @Test
+    void whenAddNullOrNotEntityAsEntityToCacheThenThrowNullPointerException() {
```

Comment:
The same here, it's hard to understand from the method name what's going on here, moreover, method name says that NullPointer should be thrown, but in the test you throw NoEntityIdException. These cases should be tested separately, even if it's one line test